### DISCUSSION TITLE
Can't see things it needs to and lacks integration, memory and attention

### DISCUSSION THREAD

**Discussion by ggb667** [community/community#120076] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Get Hub Copilot has a number of severe shortcomings that I would like to address.

* The first shortcoming is that it cannot read my file system or parts of my file system _that I designate_ - and that is a real problem. It ought to be able to see things if I tell it where those things are located.

* The second issue is that it lacks integration with GitHub itself, which is really surprising given the name of the product.  It can not see into other repositories that I have, and it can't see how the code in one particular project might relate to code in other projects and that really limits, it's use.  

* The third issue I have regards memory.  Copilot loses track of what you tell it in very short order and it has a very short attention span.  It is not playful, nor can it handle larger files (like Terraform configuration files, Maven Config files and others).  It can neither read nor write to files.  I can't have it ingest a file, adjust it and write it back.  It ought to be able to directly manipulate the IDE contents.  In addition you can't tell it to remember items for future reference.  It tells you it can't do it.  Therefore you can't tell it to always enter answer in complete examples, or to questions if it needs clarification; those sorts of things regular chat GPT can do.

**Comment by fubaduba** [comment]
Hey @ggb667  where are you using GH Copilot?
